### PR TITLE
started on thumbnail URL Util file

### PR DESCRIPTION
**Pull request by mwilkins74**

### CODE REVIEW

File: src/common/utils/thumbnailUrlUtils.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+type VideoProvider = 'youtube' | 'vimeo';
+
+/**
+* Accepts url from IVideoData as an arg and returns a tuple of the VideoProvider and the Id
+*/
+
+export function getVideoProviderAndId(videoInfo: IVideoInfo): [VideoProvider, string] {
+  const youtubeRegex = /(?:https?:\/\/)?(?:www\.)?(?:youtube\.com\/(?:[^\/\n\s]+\/\S+\/|(?:v|e(?:mbed)?)\/|\S*?[?&]v=)|youtu\.be\/)([a-zA-Z0-9_-]{11})/;
+  const vimeoRegex = /(?:https?:\/\/)?(?:www\.)?(?:player\.)?vimeo\.com\/(?:video\/)?(\d+)/;
```
Comment (ChuckBTaylor) [inline src/common/utils/thumbnailUrlUtils.ts:11]: 🔧 Define these outside of the function

File: src/common/utils/thumbnailUrlUtils.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  const youtubeMatch = videoInfo.url.match(youtubeRegex);
+  const vimeoMatch = videoInfo.url.match(vimeoRegex);
+
+  if (youtubeMatch && youtubeMatch[1]) {
+    return ['youtube', youtubeMatch[1]];
+  } else if (vimeoMatch && vimeoMatch[1]) {
+    return ['vimeo', vimeoMatch[1]];
+  }
+
+  throw new Error('Invalid video URL');
```
Comment (ChuckBTaylor) [inline src/common/utils/thumbnailUrlUtils.ts:22]: ❓ Do we want to throw or maybe just return `['unknown', null]`?

Comment (ChuckBTaylor) [reply src/common/utils/thumbnailUrlUtils.ts:22]: If we throw, we'll need some sort of catch on all consumers. We're planning to use this function when loading the video (and occaisionally thumbnails).  Something less explosive will be cleaner to work with.

**Review by kmdunn5 — APPROVED** [review]

File: src/common/utils/thumbnailUrlUtils.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+type VideoProvider = 'youtube' | 'vimeo';
+
+const youtubeRegex = /(?:https?:\/\/)?(?:www\.)?(?:youtube\.com\/(?:[^\/\n\s]+\/\S+\/|(?:v|e(?:mbed)?)\/|\S*?[?&]v=)|youtu\.be\/)([a-zA-Z0-9_-]{11})/;
+const vimeoRegex = /(?:https?:\/\/)?(?:www\.)?(?:player\.)?vimeo\.com\/(?:video\/)?(\d+)/;
+
+/**
+ * Accepts url from IVideoData as an arg and returns a tuple of the VideoProvider and the Id
+ */
+export function getVideoProviderAndId(videoInfo: IVideoInfo): [VideoProvider | 'unknown', string | null] {
```
Comment (ChuckBTaylor) [inline src/common/utils/thumbnailUrlUtils.ts:11]: 🔧 This typing is technically correct. However, I think we can make it even more clear and give the compiler even more info.  Something like this:

```suggestion
export function getVideoProviderAndId(videoInfo: IVideoInfo): [VideoProvider, string] | ['unknown', null] {
```

In this way, the compiler knows that if the `provider` is `'unknown'`, then the `id` is `null`. And vice versa, if the `provider` is not `'unknown'`, then the `id` is a non-nullable `string`

**Pull request merged**